compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -1043,11 +1043,17 @@ impl<'a, 'ra, 'tcx> BuildReducedGraphVisitor<'a, 'ra, 'tcx> {
                         span: item.span,
                     });
                 }
-                if let ItemKind::ExternCrate(Some(orig_name)) = item.kind
-                    && orig_name == kw::SelfLower
-                {
-                    self.r.dcx().emit_err(errors::MacroUseExternCrateSelf { span: attr.span });
-                }
+                match item.kind {
+                    ItemKind::ExternCrate(Some(orig_name))
+                    | ItemKind::ExternDynCrate(Some(orig_name)) => {
```

Comment:
`extern dyn crate self as foo;` should produce an error even earlier, so this probably shouldn't be necessary.

Reply:
I tried it. It is not true.

Reply:
I mean, it doesn't produce an error currently, but it should, the `dyn` semantics are not applicable to the current crate.